ckanext/datastore/blueprint.py

```diff
@@ -292,6 +305,19 @@ def dump_to(
     elif fmt == 'xml':
         writer_factory = xml_writer
         records_format = 'objects'
+    # (canada fork only): psql dump format
+    elif fmt == 'sql':
+        datastore_uri = str(DatastoreBackend.get_active_backend()._get_write_engine().url)
+        # dump clean table, schema, and data.
+        # quote all for blank values.
+        cmd = subprocess.Popen(
+            ['pg_dump', datastore_uri, '--clean', '--if-exists', '--disable-triggers',
```

Comment:
If a client requests this dump then stalls the connection this will leave the pg_dump process hung indefinitely. Need to check if this creates any locks on the DB, and maybe find a way to abort if data is transferring too slowly to prevent abuse.

Reply:
looks like we actually want to use `subprocess.run` here instead, which takes a timeout argument, and uses the `Popen` as a context manager to close any streams that it might have too. So we should be able to configure a timeout for the `pg_dump` and catch any exceptions from the `run` or the timeout raised and return an appropriate Flask response.